### DISCUSSION TITLE
RaspberryMatic 3.67.10.20230114

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2166] [Announcements]
This is release 3.67.10.20230114 of RaspberryMatic which is a **major** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.67.10.20230114/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.67.10.20230114) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=XXXXX)</sub>

## :construction: Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.65.11.20221218...3.67.10.20230114).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.67.10-3](https://github.com/jens-maus/occu/tree/b_3_67) integrating full compatibility to the [CCU3 3.67.10 firmware](https://homematic-ip.com/sites/default/files/other_media/ccu3-changelog-3-67-10.pdf) which comes with the following upstream (eQ3) performed changes:
  - With the `HmIP-SWO-*`, the parameter "Wind speed" was translated incorrectly in the English texts of the WebUI.
  - With the `HmIP-PSM*` no direct link to channel 7 was possible.
  - Device configuration parameters for channel 1 of `HmIP-eTRV-B-2` were not shown completely.
  - When creating direct links between `HmIP-MOD-RC8` and `HmIP-BBL` or `HmIP-FBL`, the correct default profile is now displayed.
  - Device images in the device firmware overview were not displayed correctly for some devices.
  - Differentiation between minimum temperature and "Off" at `eTRV-C-2` was corrected.
  - Pictures with overview of pushbutton channels for `HmIP-BS2` and `ELV-SH-BS2` corrected.
  - Incorrect display of the EULA corrected.
  - With `HmIP-FWI`, "Code 21" could not be selected in programs.
  - No negative values below "-100" could be shown in diagrams.
  - When calling the XML-RPC method "reportValueUsage" of the LegacyAPI (HmIP) with the parameter value 0, the direct link between control center and device channel was not deleted.
  - Adjustment of the slat position texts in the profile settings of direct links of blind actuators.
  - For `HmIP-eTRVs`, the status of the valve drive is now displayed under Status and operation, in the event of an error during adaptation travel. 
  - Text changed during device firmware update of event listener.
  - In the HmIP-BWTH, "Heating valve open" / "Heating valve closed" is now set under Status and operation for channel 10 depending on the "NO / NC" setting.
  - There is now a device object for the CCU3 coprocessor / RPI-RF-MOD. This means that status values such as duty cycle and carrier sense of the CCU3 coprocessor / RPI-RF-MOD can be displayed in the CCU and used in programs.
- updated the default HmIPW-DRAP firmware files to latest official version (2.4.28) available (#2145).
- modified home assistant addon config to support to also set external ports for the CCU-Jack Addon (#2011).
- implemented support to use the HomeMatic/BidCos-RF LAN-Gateway mode without having to downgrade the RF firmware but use `/dev/mmd_bidcos` as the serial device for the `hmlangw` daemon. This finally allows to use a `RPI-RF-MOD` or `HmIP-RFUSB` for the BidCos-RF LAN-Gateway mode (#2132, @jp112sdl).

#### WebUI changes:
- renamed `0078-WebUI-Fix-SetDeviceProfile` WebUI patch to `0173-WebUI-SendPOSTRequest` and unified it with changes to switch all `SendRequest()` calls in `webui.js` to a new `SendPOSTRequest()` function which will perform `Ajax.Updater()` post request calls instead so that query strings do not end up being too long for some use cases (e.g. in case of being used with an ingress proxy like in the HomeAssistant Addon case, #1189).
- fixed issue in `0155-WebUI-Add-HmIP-RFUSB-Support` WebUI patch which incorrectly removed support for the "HmIP-HAP JS1" maintenance channel support (cf. https://github.com/jens-maus/RaspberryMatic/discussions/2150#discussioncomment-4627714)
- integrated a new `0174-WebUI-Fix-MissingETRVparams` WebUI/OCCU bugfix patch which fixes an issue where since OCCU 3.65.x some options/parameters of certain HmIP-eTRV devices were not displayed anymore in the WebUI due to a upstream typo in the `legacy-parameter-definition.config` file (#2027).
- modified the home assistant ingress proxy (`ha-proxy.js`) to exclude `/jpages/` urls from the charset encoding conversion because `HMIPServer` always seems to send in utf-8 but unfortunately without explicitly adding `charset=utf-8` to the content-type. This resulted in broken charset conversions like for umlauts if used in the groups, diagrams and other HMIPServer delivered WebUI portions (#2139).
- modified `0004-LocalSSLCert` WebUI patch and `S50lighttpd` startup script to use elliptic curve based algorithms when generating default private key and https certificates.
- updated `0031-WebUI-Fix-FileUpload` WebUI bugfix patch to include a fix for accepting certificate files containing elliptic curve based private keys and certificate files (#2126).
- modified `0007-WebUI-Enhanced-FirmwareUpdate` WebUI patch to include a sha256 checksum check to first check for potential inconsistent backup files before trying to restore it. This should catch rare sbk file inconsistencies that could happen (cf. https://homematic-forum.de/forum/viewtopic.php?p=749245#p749250).
- modified `0042-WebUI-HeatingControl-HumidityBatteryValveState` WebUI patch to show the humidity also in HM/BidCos-RF heating groups (#2106, @jp112sdl).
- downgraded `prototype.js` to last 1.6 version (1.6.1) to stay compatible with the API used in the WebUI, thus fixing some issues since the last release (#2105, #609).
- integrated a regression fix for updating spectrum.js to 1.8.1 (#2102).
- updated CodeMirror to latest 5.65.11 version.

#### Operating system changes:
- updated tailscale to latest 1.34.2 version.
- updated RaspberryPi firmware+kernel to latest 1.20230106 (5.15.84).
- updated upstream kernel to 5.15.86.
- set `discard=on` as a default in the `install-proxmox.sh` VM installation script on fresh scsi0 disks to make sure the VM can properly issue trim operations via "fstrim" to decrease virtual disk size.

## :family: Contributors (alphabetically):
- @jens-maus
- @jp112sdl 

## :memo: Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## :package:  Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-rpi4.zip)<br/>SHA256: 29293c484844eb76d08b0141c8f8a9af5fbbd414af2f0bcdd5d6229f416df264</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-rpi3.zip)<br/>SHA256: 474b6815f50c95cea024d1c2676aa8e84f9ec39352874e80b0d43d1704e57489</sub>
<sub>:package: [RaspberryMatic-3.67.10.20230114-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: 1e6a77e866e12924f6fa4a3df17d1a062e26a0d9857e174ffc7019b28b44c72d</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-rpi2.zip)<br/>SHA256: ff90af96011287b9da5111192c3113f402efedb470f5db3c5f2caecb6dc833b2</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-rpi0.zip)<br/>SHA256: 8c58a7a1f070604e7932baf254ff66cd9c54f6abeaf64966839d83291e002285</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-tinkerboard.zip)<br/>SHA256: 1df00838d74b7430127908d6e3fb47084b1c8b0731aa4d02b4ce0ae589329c30</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-odroid-n2.zip)<br/>SHA256: b816f496073d83e3de4a7eac9a18e95bfe1ecdc3e7959b56c2102bd38f09a89f</sub>
<sub>:package: [RaspberryMatic-3.67.10.20230114-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-odroid-c4.zip)<br/>SHA256: 1cb43de7c530436606b2fafacba4f9672a6598f35f0773c717b158328b66b65a</sub>
<sub>:package: [RaspberryMatic-3.67.10.20230114-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-odroid-c2.zip)<br/>SHA256: 9420db8a42137af856b6238d02ed93b6fd83261e6b7c554a0276847872496fba</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-intelnuc.zip)<br/>SHA256: 01122c1e2e5c0474b2de40cd5ff37dd533e9f08607050c5f8e7301fb87096666</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-ova.zip)<br/>SHA256: da472a647029bb7c1f04e803382a33fb26a42e958f9e713b94243bf802ebb780</sub>
<sub>:package: [RaspberryMatic-3.67.10.20230114.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114.ova) (only for initial OVA installation)<br/>SHA256: 4abe63e9eb3a7c27388f6d73066d2dc52a6ad92275d7b0cc2c3025ff5e42368a</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>:package: [RaspberryMatic-3.67.10.20230114-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: 84759079310f0894726ab70a0008c61137c0a04b2b63ff49cdd92a071156fbec</sub>
<sub>:package: [RaspberryMatic-3.67.10.20230114-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: 2ddc3316d7c234ada519be31a13d9682b5cb02f562c856f0be7dab29c1426b77</sub>
<sub>:package: [RaspberryMatic-3.67.10.20230114-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.67.10.20230114/RaspberryMatic-3.67.10.20230114-oci_arm.tgz) (arm/armhf)<br/>SHA256: da773f2dfae84972c763a94be446231f8bf636a8dbfd5538c508dfcb0e824dd5</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.67.10.20230114'>RaspberryMatic 3.67.10.20230114</a>.</em>

**Comment by KaaNee** [comment]
Hi Jens, there is a firmware version missing in the changelog:
"This only applies to devices that have been CCU firmware version or higher."
